pkg/nfd-worker/nfd-worker.go

```diff
@@ -713,25 +713,21 @@ func (m *nfdWorker) updateNodeFeatureObject(labels Labels) error {
 
 	features := source.GetAllFeatures()
 
-	// Create owner ref
-	ownerRefs := []metav1.OwnerReference{}
+	// Get pod ower reference
 	podName := os.Getenv("POD_NAME")
-	podUID := os.Getenv("POD_UID")
-	if podName != "" && podUID != "" {
-		isTrue := true
-		ownerRefs = []metav1.OwnerReference{
-			{
-				APIVersion: "v1",
-				Kind:       "Pod",
-				Name:       podName,
-				UID:        types.UID(podUID),
-				Controller: &isTrue,
-			},
-		}
-	} else {
-		klog.InfoS("Cannot set NodeFeature owner reference, POD_NAME and/or POD_UID not specified")
+	client, err := m.getKubeClient()
+	if err != nil {
+		return fmt.Errorf("failed to get kube client: %w", err)
 	}
 
+	selfPod, err := client.CoreV1().Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
```

Comment:
Why not use the downward API? You pass the pod name as an envar to the worker daemonset pod template spec and retrieve it via the envar here